OWL2/ProveHermit.hs

```diff
@@ -0,0 +1,281 @@
+        {- ^ fst: identifier of the batch thread for killing it
+        snd: MVar to wait for the end of the thread -}
+hermitCMDLautomaticBatch inclProvedThs saveProblem_batch resultMVar
+                         thName defTS th freedefs =
+  genericCMDLautomaticBatch (atpFun thName) inclProvedThs saveProblem_batch
+    resultMVar hermitS thName
+    (parseTacticScript batchTimeLimit [] defTS) th freedefs emptyProofTree
+
+-- * Main prover functions
+
+{- |
+  Runs the Pellet service.
+-}
+consCheck :: String
+          -> TacticScript
+          -> TheoryMorphism Sign Axiom OWLMorphism ProofTree
+          -> [FreeDefMorphism Axiom OWLMorphism] -- ^ freeness constraints
+          -> IO (CCStatus ProofTree)
+consCheck thName (TacticScript tl) tm freedefs = case tTarget tm of
+  Theory sig nSens -> do
+    let proverStateI = owlProverState sig (toNamedList nSens) freedefs
+        prob = showOWLProblemS proverStateI
+        pStatus out tUsed = CCStatus
+          { ccResult = Nothing
+          , ccProofTree = ProofTree $ unlines out ++ "\n\n" ++ prob
+          , ccUsedTime = timeToTimeOfDay $ secondsToDiffTime $ toInteger tUsed }
+        tLim = readMaybe tl
+    res <- runTimedHermit "consistency" (basename thName ++ ".owl") prob Nothing
+      $ fromMaybe maxBound $ readMaybe tl
+    return $ case res of
+      Nothing -> pStatus ["Timeout after " ++ tl ++ " seconds"]
+                 (fromMaybe (0 :: Int) tLim)
+      Just (progTh, outp, eOut) -> if progTh then
+          let (_, exCode, out, tUsed) = analyseOutput outp eOut
+          in (pStatus out tUsed) { ccResult = exCode }
+        else pStatus ["Hermit not found"] (0 :: Int)
+
+runTimedHermit :: String -- ^ pellet subcommand
+  -> FilePath            -- ^ basename of problem file
+  -> String              -- ^ problem content
+  -> Maybe String        -- ^ entail content
+  -> Int                 -- ^ time limit in seconds
+  -> IO (Maybe (Bool, String, String)) -- ^ timeout or (success, stdout, stderr)
+runTimedHermit opts tmpFileName prob entail secs = do
+  (progTh, pPath) <- check4jarFile hermitEnv hermitJar
+  if progTh then withinDirectory pPath $ do
+      timeTmpFile <- getTempFile prob tmpFileName
+      timeTmpEnt <- getTempFile (fromMaybe "" entail) tmpFileName
+      let entFile = timeTmpFile ++ ".entail.owl"
+          doEntail = trace (show opts) (isJust entail)
+          args = "-Xmx1024M" : "-jar" : hermitJar
+           : (if doEntail then ["--premise=file://"++timeTmpFile, "--conclusion=file://"++entFile, "-E"] 
+              else ["-k", timeTmpFile]) ++ words opts
+      case entail of
+        Just c -> writeFile entFile c
+        Nothing -> return ()
+      mex <- timeoutCommand secs "java" args
+      trace (show mex ++ "\n" ++ show args) (removeFile timeTmpFile)
+      removeFile timeTmpEnt
+      when doEntail $ removeFile entFile
+      return $ fmap (\ (_, outS, errS) -> if outS == "true\n" then (True, outS, errS) else (False, outS, errS)) mex
```

Comment:
Using False here make the tool report it cannot find the prover. It worked for me as expected with:
`return $ fmap (\ (_, outS, errS) -> (True, outS, errS)) mex`